backend/retentionpolicy.ts

```diff
@@ -0,0 +1,180 @@
+import { model } from 'mongoose'
+import {
+  AnyState,
+  ExpenseReport as IExpenseReport,
+  HealthCareCost as IHealthCareCost,
+  Travel as ITravel,
+  User as IUser,
+  Locale,
+  ReportType,
+  reportIsHealthCareCost,
+  reportIsTravel,
+  schemaNames
+} from '../common/types.js'
+import i18n from './i18n.js'
+import { sendMail } from './mail/mail.js'
+import Settings from './models/settings.js'
+import User from './models/user.js'
+
+async function getForRetentionPolicy(schema: schemaNames, date: Date, state: AnyState) {
+  return await model(schema).find({ state: state, updatedAt: { $lt: date }, historic: false })
+}
+
+function getDateThreshold(days: number) {
+  let dateThreshold = new Date()
+  dateThreshold.setHours(0, 0, 0, 0)
+  dateThreshold.setDate(dateThreshold.getDate() - days)
+  return dateThreshold
+}
+
+async function triggerDeletion(retentionSettings: {
+  deleteRefundedAfterXDays: number
+  deleteApprovedTravelAfterXDaysUnused: number
+  deleteInWorkReportsAfterXDaysUnused: number
+}) {
+  let deletions: { schema: schemaNames; state: AnyState; deletionPeriod: number }[] = [
+    { schema: 'Travel', state: 'refunded', deletionPeriod: retentionSettings.deleteRefundedAfterXDays },
+    { schema: 'Travel', state: 'approved', deletionPeriod: retentionSettings.deleteApprovedTravelAfterXDaysUnused },
+    { schema: 'ExpenseReport', state: 'refunded', deletionPeriod: retentionSettings.deleteRefundedAfterXDays },
+    { schema: 'ExpenseReport', state: 'inWork', deletionPeriod: retentionSettings.deleteInWorkReportsAfterXDaysUnused },
+    { schema: 'HealthCareCost', state: 'refunded', deletionPeriod: retentionSettings.deleteRefundedAfterXDays },
+    { schema: 'HealthCareCost', state: 'inWork', deletionPeriod: retentionSettings.deleteInWorkReportsAfterXDaysUnused }
```

Comment:
Ist ja das gleiche wie in Zeile 71-77. Überleg mal wie man hinkriegst das nur einmal zu schreiben.